stories/demos/LegacyGrouping.tsx

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, { useState, useMemo } from 'react';
+import groupBy from 'lodash/groupBy';
+
+import { GroupRowRenderer, RowData, RowExpandToggleEvent } from './components/RowRenderers';
+import DataGrid, { Column } from '../../src';
+
+interface Row extends RowData<Row, unknown> {
+  id: number;
+  task: string;
+  complete: number;
+  priority: string;
+  issueType: string;
+}
+
+const columns: Column<any>[] = [
+  {
+    key: 'id',
+    name: 'ID',
+    width: 80
+  },
+  {
+    key: 'task',
+    name: 'Title'
+  },
+  {
+    key: 'priority',
+    name: 'Priority'
+  },
+  {
+    key: 'issueType',
+    name: 'Issue Type'
+  },
+  {
+    key: 'complete',
+    name: '% Complete'
+  }
+];
+
+function createRows(): Row[] {
+  const rows = [];
+  for (let i = 1; i < 500; i++) {
+    rows.push({
+      id: i,
+      task: `Task ${i}`,
+      complete: Math.min(100, Math.round(Math.random() * 110)),
+      priority: ['Critical', 'High', 'Medium', 'Low'][Math.floor((Math.random() * 3) + 1)],
+      issueType: ['Bug', 'Improvement', 'Epic', 'Story'][Math.floor((Math.random() * 3) + 1)]
+    });
+  }
+
+  return rows;
+}
+
+function groupByColumn(rows: Row[], columnKeys: string[], expandedGroups: Set<string>, treeDepth = 0, parentKey = '') {
+  if (columnKeys.length === 0) return rows;
+  const gridRows: any = [];
+  const [columnKey, ...remainingColumnKeys] = columnKeys;
+  const groupedRows = groupBy(rows, columnKey);
+  const groupedKeys = Object.keys(groupedRows);
+  for (const key of groupedKeys) {
+    const groupKey = parentKey ? `${parentKey}_${key}` : key;
+    const isExpanded = expandedGroups.has(groupKey);
+    const rowGroupHeader = {
+      groupKey,
+      name: key,
+      __metaData: {
```

Comment:
I think it'd be better to refactor this to a more realistic data structure, maybe something like
```ts
interface RowGroup {
  type: 'group';
  // ...
}

interface RowData {
  type: 'data';
  // ...
}

type Row = RowGroup | RowData;
```

Not worth our time right now though.

Reply:
Simplified types a bit